### PR TITLE
Added fullname to export

### PR DESCRIPTION
**Pull request by martinyde**
#### Description

Add fullname to exports

### CODE REVIEW

File: Services/TimesheetsExporter.php [hunk 1]
```diff
@@ -26,18 +26,36 @@ public function __construct(
...
+      $invEmpl = filter_var($criteria[ 'invEmpl' ] ?? NULL, FILTER_VALIDATE_BOOLEAN) ? '1' : '';
+      $invComp = filter_var($criteria[ 'invComp' ] ?? NULL, FILTER_VALIDATE_BOOLEAN) ? '1' : '';
+      $paid = filter_var($criteria[ 'paid' ] ?? NULL, FILTER_VALIDATE_BOOLEAN) ? '1' : '';
+      $clientId = (int) ($criteria[ 'clientId' ] ?? 0);
+
+      $data = $this->timesheets->getAll($projectId, $kind, $dateFrom, $dateTo, $userId, $invEmpl, $invComp, $paid, $clientId, 0);
+
+      foreach ($data as $key => $row) {
+        $data[$key]['fullname'] = $this->addFullname($row);
+      }
```
Comment (rimi-itk) [inline Services/TimesheetsExporter.php]: You don't need the `$key` variable here:

```suggestion
      foreach ($data as&$row) {
        $row['fullname'] = $this->addFullname($row);
      }
```

File: Services/TimesheetsExporter.php [hunk 2]
```diff
@@ -26,18 +26,36 @@ public function __construct(
...
+      $userId = (int) ($criteria[ 'userId' ] ?? 0);
+      $invEmpl = filter_var($criteria[ 'invEmpl' ] ?? NULL, FILTER_VALIDATE_BOOLEAN) ? '1' : '';
+      $invComp = filter_var($criteria[ 'invComp' ] ?? NULL, FILTER_VALIDATE_BOOLEAN) ? '1' : '';
+      $paid = filter_var($criteria[ 'paid' ] ?? NULL, FILTER_VALIDATE_BOOLEAN) ? '1' : '';
+      $clientId = (int) ($criteria[ 'clientId' ] ?? 0);
+
+      $data = $this->timesheets->getAll($projectId, $kind, $dateFrom, $dateTo, $userId, $invEmpl, $invComp, $paid, $clientId, 0);
+
+      foreach ($data as $key => $row) {
+        $data[$key]['fullname'] = $this->addFullname($row);
```
Comment (rimi-itk) [inline Services/TimesheetsExporter.php]: `addFullname` should be called `computeFullname` (or something similar). And I think it's superfluous:

```suggestion
        $data[$key]['fullname'] = $row['firstname'].' '.$row['lastname'];
```

File: Services/TimesheetsExporter.php [hunk 3]
```diff
@@ -26,18 +26,36 @@ public function __construct(
...
-        $dateFrom = $this->getDateTime($criteria['dateFrom'] ?? null, new Carbon('2000-01-01'));
-        $dateTo = $this->getDateTime($criteria['dateTo'] ?? null, new Carbon('2100-01-01'));
-        $userId = (int) ($criteria['userId'] ?? 0);
-        $invEmpl = filter_var($criteria['invEmpl'] ?? null, FILTER_VALIDATE_BOOLEAN) ? '1' : '';
-        $invComp = filter_var($criteria['invComp'] ?? null, FILTER_VALIDATE_BOOLEAN) ? '1' : '';
-        $paid = filter_var($criteria['paid'] ?? null, FILTER_VALIDATE_BOOLEAN) ? '1' : '';
-        $clientId = (int) ($criteria['clientId'] ?? 0);
-
-        return $this->timesheets->getAll($projectId, $kind, $dateFrom, $dateTo, $userId, $invEmpl, $invComp, $paid, $clientId, 0);
+    public function generateData(array $criteria): array {
```
Comment (rimi-itk) [inline Services/TimesheetsExporter.php]: Remember the coding standards: https://github.com/ITK-Leantime/leantime-dataexport?tab=readme-ov-file#coding-standards

**Review by rimi-itk — CHANGES_REQUESTED** [review]

File: Services/TimesheetsExporter.php (hunk 1 above)
Comment (martinyde) [reply Services/TimesheetsExporter.php]: fixed

File: Services/TimesheetsExporter.php (hunk 3 above)
Comment (martinyde) [reply Services/TimesheetsExporter.php]: Added

File: Services/TimesheetsExporter.php (hunk 2 above)
Comment (martinyde) [reply Services/TimesheetsExporter.php]: Removed

Comment (martinyde) [reply Services/TimesheetsExporter.php]: Removed method

**Review by rimi-itk — APPROVED** [review]

**Pull request merged**